src/main/java/com/jeovan/gymcrmsystem/daos/TraineeDao.java

```diff
@@ -1,15 +1,22 @@
 package com.jeovan.gymcrmsystem.daos;
 
 import com.jeovan.gymcrmsystem.models.Trainee;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
 
-import java.util.Map;
+import java.util.Optional;
 import java.util.UUID;
 
 @Component
-public class TraineeDao extends SimpleDao<Trainee>{
+public interface TraineeDao extends JpaRepository<Trainee, UUID> {
 
-    public TraineeDao(Map<UUID, Trainee> storage) {
-        super(storage);
-    }
+    @Query("SELECT t FROM Trainee t JOIN t.user u WHERE u.username = ?1")
```

Comment:
Do you have any reason to use @Query when you can just use[ query name method](https://docs.spring.io/spring-data/jpa/reference/jpa/query-methods.html) and leave that spring create the query for you?

Reply:
This was very silly, I was not aware I could use deleteByUserUsername and avoid having to create a Join, this was fixed in the next commit